src/utils/strutils.c

```diff
@@ -268,14 +271,19 @@ int strnquote(char *dest, size_t size, const char *s, int n,
     if (size > i) dest[i] = '"';
     i++;
   }
-  while (s[j] && (n < 0 || (int)j < n)) {
-    if (s[j] == '"' && !(flags & strquoteNoEscape)) {
-      if (size > i) dest[i] = '\\';
+  if (s) {
```

Comment:
hard to follow the logic. Comment line-by-line what is happening here, or simplify the expression with standard functions.

Reply:
Added some comments to make it easier to follow.